src/Microsoft.DotNet.ProjectModel/ProjectContext.cs

```diff
@@ -154,6 +155,100 @@ public static IEnumerable<ProjectContext> CreateContextForEachTarget(string proj
                         .BuildAllTargets();
         }
 
+        public static bool IsRidCompatibleWith(string rid, string otherRid)
```

Comment:
What if we create a dict of rids -> compatible rids.

This won't work for example for `aot` rids in netcore50